### ISSUE TITLE
Relogging as Spectator disables noclip

### ISSUE THREAD

**Issue by suresttexas00** [benbaptist/minecraft-wrapper#354]
@JasonBristol reported:
"This is a legacy issue that has existed where a user disconnecting in spectator mode will not be able to clip through objects until they are reissued spectator mode.

Potentially related to the cache?"

**Comment by suresttexas00** [comment]
I have narrowed this to something we are doing incorrectly with the playerlistitem packet.  I suspect this note applies:

```
  # Action Update Gamemode
                elif action == 1:
                    data = self.packet.readpkt([VARINT])
                    # I think if we are sending this packet to our own player
                    # we need to send the client the online UUID.  Everyone
                    # else would get the offline version
                    # TODO Corrects the noclip spectator problem??
                    gamemode = data[0]
                    # ("varint:gamemode")

                    self.client.packet.sendpkt(
                        self.pktCB.PLAYER_LIST_ITEM[PKT],
                        [VARINT, VARINT, UUID, VARINT],
                        (1, 1, uuid_to_send, gamemode), serverbound=False)
```

**Comment by suresttexas00** [comment]
I put the wrapper in offline mode (to simplify things):

with wrapper parsing of the playerlistitem, the server sends this and the player can go through walls:
```
<<==CB3 (id_46)PLAYER_LIST_ITEM-0x2e: 'x_2i²[Θ<*»lzvxxrSurestTexas00x_x_x_'
```
When wrapper parses the packet, it looks like this and the player cannot noclip and (oddly) it sends two of them:
```
<<==CB (PROXY-MOD)(46)PLAYER_LIST_ITEM-0x2e: '2i²[Θ<*»lzvxxrSurestTexas00x_x_x_x_'
<<==CB (PROXY-MOD)(46)PLAYER_LIST_ITEM-0x2e: '2i²[Θ<*»lzvxxrSurestTexas00x_x_x_x_'
```

**Comment by suresttexas00** [comment]
Our hacky parse of player list flatly installs "0" (gamemode survival) to the player list gamemodes (the `�` above that is different is `0x03`). It sends this for our gamemode:
```
raw += self.client.packet.send_varint(0)
```

Our parser also splits muliple playerlist items into several playerlist packets with one entry each.

Total rewrite will be done here...

**Comment by suresttexas00** [comment]
This is also why player lists never show the GM3 player as _italicized_.

**Comment by suresttexas00** [comment]
Fixed completely build 33 version 1.0b33
